env.go

```diff
@@ -160,29 +214,65 @@ func getEcrToken() (username, password string, err error) {
 		return
 	}
 	for _, authData := range output.AuthorizationData {
-		// authData.AuthorizationToken is a base64-encoded username:password string,
-		// where the username is always expected to be "AWS".
+		if b, err := strconv.ParseBool(os.Getenv(envDebugMode)); err == nil && b {
+			if authData.ExpiresAt != nil {
+				expiration := authData.ExpiresAt.UTC().Format(time.RFC3339)
+				_, _ = fmt.Fprintf(os.Stderr, "ECR token for %q will expire at %s (UTC)\n", account, expiration)
+			}
+		}
+
+		if authData.AuthorizationToken == nil {
+			err = fmt.Errorf("ecr: authorization token for %q is nil", account)
+			return
+		}
+
 		var tokenBytes []byte
 		tokenBytes, err = base64.StdEncoding.DecodeString(*authData.AuthorizationToken)
 		if err != nil {
 			return
 		}
 		token := bytes.SplitN(tokenBytes, []byte{':'}, 2)
+		if len(token) != 2 {
+			err = fmt.Errorf("ecr: invalid authorization token format for %q", account)
+			return
+		}
+
 		username, password = string(token[0]), string(token[1])
 	}
 	return
 }
 
-func getRoleArn(configSources ...interface{}) (roleARN string) {
+// getRoleArn retrieves the AWS role ARN for a specific account by checking environment variables and AWS configurations.
+// It checks the account-specific role ARN environment variable (AWS_ROLE_ARN_<account>). If not found,
+// then checks the standard AWS role ARN environment variable (AWS_ROLE_ARN) when no config sources are provided.
+// Finally, checks config sources which may contain role ARNs in AWS environment config or shared config.
+// Returns role ARN string if found, empty string otherwise.
+func getRoleArn(account string, configSources ...any) (roleARN string, err error) {
+	val, found := os.LookupEnv(envAwsRoleArn + "_" + account)
+	if found {
+		return strings.TrimSpace(val), nil
+	}
+
+	// Check if any account-specific AWS credentials exist
+	_, hasAccessKey := os.LookupEnv(envAwsAccessKeyID + "_" + account)
+	_, hasSecretKey := os.LookupEnv(envAwsSecretAccessKey + "_" + account)
+	if hasAccessKey || hasSecretKey {
+		return "", fmt.Errorf("account-specific environment variables for %q are set, but no role ARN found", account)
```

Comment:
It's probably fine to not provide a role ARN with suffixed credentials, we just need to be sure not to try and assume the non-suffixed role. So I don't think we need to add the error return to the signature?

```suggestion
		return ""
```